### PR TITLE
US011 add view models

### PR DESCRIPTION
**Pull request by SamDotNetDev**

### CODE REVIEW

File: ParkingZoneApp/ViewModels/CreateVM.cs [hunk 1]
```diff
@@ -0,0 +1,19 @@
+﻿using ParkingZoneApp.Models;
+
+namespace ParkingZoneApp.ViewModels
+{
+    public class CreateVM
```
Comment (abdullokhtolibjonov) [inline ParkingZoneApp/ViewModels/CreateVM.cs]: put the VMs inside ParkingZone Folder ViewModels/ParkingZone/CreateVM

File: ParkingZoneApp/ViewModels/DeleteVM.cs [hunk 2]
```diff
@@ -0,0 +1,24 @@
+﻿
+using ParkingZoneApp.Models;
+
+namespace ParkingZoneApp.ViewModels
+{
+    public class DeleteVM
```
Comment (abdullokhtolibjonov) [inline ParkingZoneApp/ViewModels/DeleteVM.cs]: i said yesterday we dont need deleteVM

File: ParkingZoneApp/ViewModels/DetailsVM.cs [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+namespace ParkingZoneApp.ViewModels
+{
+    public class DetailsVM
+    {
+        public int Id { get; set; }
+        public string Name { get; set; }
+        public string Address { get; set; }
+
+        public DetailsVM MapToModel(ParkingZone parkingZone)
```
Comment (abdullokhtolibjonov) [inline ParkingZoneApp/ViewModels/DetailsVM.cs]: here you are not doing MapToModel, but MapToViewModel()

Comment (abdullokhtolibjonov) [inline ParkingZoneApp/ViewModels/DetailsVM.cs]:
```suggestion
        public DetailsVM MapToVM(ParkingZone parkingZone)
```

File: ParkingZoneApp/ViewModels/EditVM.cs [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+namespace ParkingZoneApp.ViewModels
+{
+    public class EditVM
+    {
+        public int Id { get; set; }
+        public string Name { get; set; }
+        public string Address { get; set; }
+        public DateTime DateOfEstablishment {  get; set; }
+
+        public EditVM MapToModel(ParkingZone parkingZone)
```
Comment (abdullokhtolibjonov) [inline ParkingZoneApp/ViewModels/EditVM.cs]:
```suggestion
        public EditVM MapToVM(ParkingZone parkingZone)
```

File: ParkingZoneApp/ViewModels/IndexVM.cs [hunk 5]
```diff
@@ -0,0 +1,17 @@
+﻿using ParkingZoneApp.Models;
+
+namespace ParkingZoneApp.ViewModels
+{
+    public class IndexVM
```
Comment (abdullokhtolibjonov) [inline ParkingZoneApp/ViewModels/IndexVM.cs]:
```suggestion
    public class ListItemVM
```

File: ParkingZoneApp/ViewModels/IndexVM.cs [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+    {
+        public IEnumerable<ParkingZone> ParkingZones { get; set; }
+
+        public IndexVM MapToModel(IEnumerable<ParkingZone> parkingZones)
```
Comment (abdullokhtolibjonov) [inline ParkingZoneApp/ViewModels/IndexVM.cs]:
```suggestion
        public IndexVM MapToVM(IEnumerable<ParkingZone> parkingZones)
```

File: ParkingZoneApp/Areas/Admin/Controllers/ParkingZoneController.cs [hunk 7]
```diff
@@ -21,7 +22,7 @@ public ParkingZoneController(IParkingZoneService service)
         public IActionResult Index()
         {
             var parkingZones = _service.GetAll();
-            return View(parkingZones);
+            return View(new IndexVM().MapToModel(parkingZones));
```
Comment (abdullokhtolibjonov) [inline ParkingZoneApp/Areas/Admin/Controllers/ParkingZoneController.cs]: dont do like that create variable and assign to it. Then return

File: ParkingZoneApp/Areas/Admin/Controllers/ParkingZoneController.cs [hunk 8]
```diff
@@ -32,8 +33,8 @@ public IActionResult Details(int? id)
             {
                 return NotFound();
             }
-
-            return View(parkingZone);
+            
+            return View(new DetailsVM().MapToModel(parkingZone));
```
Comment (abdullokhtolibjonov) [inline ParkingZoneApp/Areas/Admin/Controllers/ParkingZoneController.cs]: dont do like that

File: ParkingZoneApp/Areas/Admin/Controllers/ParkingZoneController.cs [hunk 9]
```diff
@@ -91,7 +91,7 @@ public IActionResult Edit(int id, ParkingZone parkingZone)
                 }
                 return RedirectToAction(nameof(Index));
             }
-            return View(parkingZone);
+            return View(new EditVM().MapToModel(parkingZone));
```
Comment (abdullokhtolibjonov) [inline ParkingZoneApp/Areas/Admin/Controllers/ParkingZoneController.cs]: for Post Edit VM should come

File: ParkingZoneApp/Areas/Admin/Controllers/ParkingZoneController.cs [hunk 10]
```diff
@@ -102,8 +102,7 @@ public IActionResult Delete(int? id)
             {
                 return NotFound();
             }
-
-            return View(parkingZone);
+            return View(new DeleteVM().MapToModel(parkingZone));
```
Comment (abdullokhtolibjonov) [inline ParkingZoneApp/Areas/Admin/Controllers/ParkingZoneController.cs]: do not do like that

File: ParkingZoneApp/Areas/Admin/Views/ParkingZone/Details.cshtml [hunk 11]
```diff
@@ -24,12 +24,12 @@
...
             @Html.DisplayFor(model => model.Address)
         </dd>
-        <dt class="col-sm-12">
+        @* <dt class="col-sm-12">
             <label>Creation time</label>
         </dt>
         <dd class="col-sm-12">
             @Html.DisplayFor(model => model.DateOfEstablishment)
-        </dd>
+        </dd> *@
```
Comment (abdullokhtolibjonov) [inline ParkingZoneApp/Areas/Admin/Views/ParkingZone/Details.cshtml]: remove

File: ParkingZoneApp/Areas/Admin/Views/ParkingZone/Create.cshtml [hunk 12]
```diff
@@ -23,11 +23,11 @@
                 <input asp-for="Address" class="form-control" />
                 <span asp-validation-for="Address" class="text-danger"></span>
             </div>
-            <div class="form-group">
+            @* <div class="form-group">
                 <label asp-for="DateOfEstablishment" class="control-label">Creation time</label>
                 <input asp-for="DateOfEstablishment" class="form-control"/>
                 <span asp-validation-for="DateOfEstablishment" class="text-danger"></span>
-            </div>
+            </div> *@
```
Comment (abdullokhtolibjonov) [inline ParkingZoneApp/Areas/Admin/Views/ParkingZone/Create.cshtml]: remove comments

**Review by abdullokhtolibjonov — CHANGES_REQUESTED** [review]

File: ParkingZoneApp/Areas/Admin/Controllers/ParkingZoneController.cs [hunk 13]
```diff
@@ -21,7 +21,14 @@ public ParkingZoneController(IParkingZoneService service)
         public IActionResult Index()
         {
             var parkingZones = _service.GetAll();
-            return View(parkingZones);
+            var listItemVMs = parkingZones.Select(parkingZone => new ListItemVM
```
Comment (abdullokhtolibjonov) [inline ParkingZoneApp/Areas/Admin/Controllers/ParkingZoneController.cs]: move this logic into vm class

File: ParkingZoneApp/Areas/Admin/Controllers/ParkingZoneController.cs [hunk 14]
```diff
@@ -63,15 +70,15 @@ public IActionResult Edit(int? id)
             {
                 return NotFound();
             }
-
-            return View(parkingZone);
+            var VM = new EditVM().MapToVM(parkingZone);
```
Comment (dotnetdew) [inline ParkingZoneApp/Areas/Admin/Controllers/ParkingZoneController.cs:73]: instead of it lets make constructor

**Review by abdullokhtolibjonov — CHANGES_REQUESTED** [review]

File: ParkingZoneApp/Areas/Admin/Controllers/ParkingZoneController.cs (hunk 14 above)
Comment (abdullokhtolibjonov) [inline ParkingZoneApp/Areas/Admin/Controllers/ParkingZoneController.cs]:
```suggestion
            var VM = new EditVM(parkingZone);
```

Comment (abdullokhtolibjonov) [reply ParkingZoneApp/Areas/Admin/Controllers/ParkingZoneController.cs]: lets make like that constructor

**Review by abdullokhtolibjonov — CHANGES_REQUESTED** [review]

**Review by abdullokhtolibjonov — APPROVED** [review]

**Pull request merged**